cobigen/cobigen-core-parent/cobigen-core/src/main/java/com/devonfw/cobigen/impl/util/ConfigurationUtil.java

```diff
@@ -0,0 +1,112 @@
+package com.devonfw.cobigen.impl.util;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.net.URI;
+import java.nio.charset.Charset;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Properties;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.log4j.Logger;
+
+import com.devonfw.cobigen.api.constants.ConfigurationConstants;
+import com.devonfw.cobigen.api.util.CobiGenPathUtil;
+
+/**
+ * Utilities related to the cobigen configurations including:
+ *
+ * 1. templates location
+ */
+public class ConfigurationUtil {
+
+    private static final Logger LOG = Logger.getLogger(ConfigurationUtil.class);
+
+    /**
+     * The method finds location of templates. It could be CobiGen_Templates folder or a template artifact
+     * @return template location uri if exist, otherwise null
+     */
+    public static URI findTemplatesLocation() {
+        Path configFile = getConfigurationFile();
+        if (configFile != null && Files.exists(configFile)) {
+            LOG.info("Configuration file is located at " + configFile);
+            Properties props = readConfigrationFile(configFile);
+            String templatesLocation = props.getProperty(ConfigurationConstants.COBIGEN_CONFIG_TEMPLATES_LOCATION_KEY);
+            if (StringUtils.isNotEmpty(templatesLocation)) {
+                Path templatesPath = Paths.get(templatesLocation);
+                if (Files.exists(templatesPath)) {
+                    return Paths.get(templatesLocation).toUri();
+                } else {
+                    LOG.info(
+                        "Could not find the given templates location in configuration file. Try to look for templates in cobigen home");
+                }
+            } else {
+                LOG.info(
+                    "No templates location is found in configuration file. Try to look for templates in cobigen home");
```

Comment:
Better indicate the property which is not set to teach the user on how to change the behavior.
For example `LOG.info("Property {} not set in {}. Trying to look for templates in cobigen home directory {}", ConfigurationConstants.COBIGEN_CONFIG_TEMPLATES_LOCATION_KEY, configFile, <variable indicating cobigen home dir>);`